packetinjector/inject_packet.go

```diff
@@ -170,16 +171,22 @@ func InjectPackets(pp *PacketInjectionParams, g *graph.Graph, chnl *channels) (s
 	if err != nil {
 		return "", err
 	}
+	trackID, _ := DoInjections(pp, chnl, injector, forgePackets, tid)
+	return trackID, nil
+}
 
+// DoInjections inject the packets
+func DoInjections(pp *PacketInjectionParams, chnl *Channels, injector *PacketInjector, forgePackets PacketForger, tid string) (string, chan bool) {
 	p := make(chan bool)
+	done := make(chan bool)
```

Comment:
The idea was to use a chan of injected packet

```
injected := make(chan, struct{})
```

so that line 210 you publish a message on the chan. The caller can read this chan like this:

```
for range injected {
   fmt.Printf("New packet injected")
}

fmt.Printf("Injection done")
```

The user can see the progression of the injection, even adding a counter caller side

when the injection is done, you just need to close the chan.

Reply:
with this the agent injector able to inject only one packet,
so i added this in agent side.

```
go func(injected chan string) {
		for range injected {
		}
	}(injected)
```